### PR TITLE
refactor(): package & specific version download

### PR DESCRIPTION
**Pull request by jkuri**
note that `storage` is currently not in use and will be refactored (probably with RxJS) in the next PR.

### CODE REVIEW

File: src/api/routes.ts [hunk 1]
```diff
@@ -2,10 +2,14 @@ import * as express from 'express';
...
 import { getConfig, getConfigPath, getFilePath } from './utils';
-import { writeJsonFile } from './fs';
+import { writeJsonFile, exists, existsSync, ensureDirectory, writeFile, readJsonFile } from './fs';
 import { Package } from './package';
 import * as proxy from './proxy';
 import { storage, findPackage } from './storage';
+import { createReadStream, createWriteStream } from 'fs';
+import * as request from 'request';
+import { dirname } from 'path';
+import { createGzip } from 'zlib';
```
Comment (Izak88) [inline src/api/routes.ts]: `createGzip`, `existsSync` and `writeFile` are never used.

File: src/api/proxy.ts [hunk 2]
```diff
@@ -1,88 +1,42 @@
 import * as request from 'request';
-import { createWriteStream } from 'fs';
 import { dirname } from 'path';
-import { writeJsonFile, ensureDirectory, exists } from './fs';
+import { writeJsonFile, ensureDirectory, exists, writeFile } from './fs';
 import { getFilePath, getConfig } from './utils';
 import { INpmPackage } from './package';
 import { findPackage, storage, updatePkgStorage } from './storage';
 import * as logger from './logger';
+import { createReadStream, createWriteStream } from 'fs';
```
Comment (Izak88) [inline src/api/proxy.ts]: Only request, logger and `INpmPackage` have to be imported here.

**Review by Izak88 — COMMENTED** [review]
Looks nice, some unused imports, but nothing serious...

File: src/api/proxy.ts (hunk 2 above)
Comment (jkuri) [reply src/api/proxy.ts]: updated.

**Review by Izak88 — APPROVED** [review]

**Pull request merged**